app/components/primer/octicon_symbol_component.rb

```diff
@@ -0,0 +1,60 @@
+# frozen_string_literal: true
+
+require "octicons"
+
+module Primer
+  # OcticonSymbol renders a symbol dictionary using a list of <%= link_to_octicons %>.
+  class OcticonSymbolComponent < Primer::Component
+    # Required list of icons. These will be the icons that are rendered.
+    #
+    # @param symbol [String] Name of <%= link_to_octicons %> to use.
+    # @param size [Symbol] <%= one_of(Primer::OcticonComponent::SIZE_MAPPINGS) %>
+    renders_many :icons, "Symbol"
+
+    # @example Symbol dictionary
+    #   <%= render(Primer::OcticonComponent.new(icon: :check, use_symbol: true, color: :icon_success)) %>
+    #   <%= render(Primer::OcticonComponent.new(icon: :check, use_symbol: true, color: :text_danger)) %>
+    #   <%= render(Primer::OcticonComponent.new(icon: :check, use_symbol: true, size: :medium)) %>
+    #   <%= render(Primer::OcticonSymbolComponent.new) do |c| %>
+    #     <%= c.icon(symbol: :check) %>
```

Comment:
My gut says that we might be better off passing the icons in as a hash, as we'll be instantiating an object for each icon in this case.